app/controllers/web_requests_controller.rb

```diff
@@ -24,7 +26,13 @@ def handle_request
     if user
       agent = user.agents.find_by_id(params[:agent_id])
       if agent
-        content, status, content_type = agent.trigger_web_request(request)
+        content, status, content_type, headers = agent.trigger_web_request(request)
+
+        if headers.presence
+          headers.each do |k,v|
+            response.headers[k] = v
+          end
+        end
```

Comment:
I think people who are editing agents are generally trusted users, but I'm not expert enough on what can be done with all the different HTTP headers that could be set. Which ones would you blacklist?